pkg/apic/apiserviceinstance.go

```diff
@@ -71,8 +83,62 @@ func (c *ServiceClient) updateAPIServiceInstance(
 	return instance
 }
 
+func (c *ServiceClient) createOrUpdateAccessRequestDefinition(data *mv1a.AccessRequestDefinition) (*mv1a.AccessRequestDefinition, error) {
+	// TODO - check cache for access request, update if needed
+	ardBytes, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf(c.cfg.GetEnvironmentURL() + "/accessrequestdefinitions")
+
+	response, err := c.ExecuteAPI(coreapi.POST, url, nil, ardBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	newARD := &mv1a.AccessRequestDefinition{}
+	err = json.Unmarshal(response, newARD)
+	if err != nil {
+		return nil, err
+	}
+
+	return newARD, nil
+}
+
+func (c *ServiceClient) createAccessRequestDefintion(serviceBody *ServiceBody) error {
```

Comment:
Should this be an internal processing in ServiceBody?

Reply:
Moved, called as part of the builder